src/Gaffer/Reference.cpp

```diff
@@ -240,9 +241,12 @@ void Reference::loadInternal( const std::string &fileName )
 	// exception that we throw.
 
 	bool errors = false;
-	if( !fileName.empty() )
+	const char *s = getenv( "GAFFER_REFERENCE_PATHS" );
+	IECore::SearchPath sp( s ? s : "" );
+	boost::filesystem::path path = sp.find( fileName );
+	if( !path.empty() )
```

Comment:
This check means that we no longer throw an exception if the file can't be found, which unfortunately we didn't have a test case for. Here's one that shows the problem :

```
	def testLoadThrowsOnMissingFile( self ) :

		s = Gaffer.ScriptNode()
		s["r"] = Gaffer.Reference()

		with self.assertRaisesRegexp( Exception, "I/O Exception : Unable to open file \"thisFileDoesntExist.grf\"" ) :
			s["r"].load( "thisFileDoesntExist.grf" )
```

Could you add that to this PR and fix the problem? I think we still need the `fileName.empty()` check, because that's used when undoing an initial `load()` call, which will result in `loadInternal( "" )`.

Reply:
I initially had it throw from inside `loadInternal()` but it broke the `testUndo()` case on line 1000. I should have been more curious about that.

This latest commit puts a search check and throw in the `load()` which passes all three tests for me.

Thanks